src/m_misc.c

```diff
@@ -83,6 +83,72 @@ boolean M_FileExists(char *filename)
     }
 }
 
+char *M_FileCaseExists(char *path)
```

Comment:
Oh. Then please change it to always return a newly-allocated string. Simpler function semantics are better/safer.